sphinx/source/docs/dev_guide/creating_pull_requests.rst

```diff
@@ -0,0 +1,153 @@
+
+    First, pick an issue from Bokeh's `issue tracker`_. Issues that are
+    relatively easy to get started on are labeled `good first issue`_.
+
+    Next, post a comment in the issue's discussion to indicate that you are
+    working on this issue. If there is no issue yet, you should first
+    :ref:`create one <devguide_issues>`.
+
+2. Create a branch
+    Before writing any code, you need to create a new `branch`_ on
+    :ref:`your fork <devguide_setup_cloning>` of the Bokeh repository. In most
+    cases, you should base your new branch on Bokeh's ``default`` branch. The
+    default branch is usually a branch that represents the next version of
+    Bokeh, for example ``branch-2.4``.
+
+    Use this pattern to name your branch:
+
+    ``[issue number]_[Python/BokehJS model you are working on, if applicable]_[optional: model property or short description]``
+
+    For example:
+
+    ``11423_table_column_visible``
+
+    See `BEP 6: Branching Strategy`_ for more information on the different
+    branches in the Bokeh repository.
+
+3. Publish your branch and push a first commit
+    Before you can open a pull request, you first need to publish your branch.
+    At this time, you should also include a first commit.
+
+4. Open a new pull request
+    After publishing your branch and adding a first commit, go to the
+    `Bokeh repository`_ on GitHub. GitHub should have detected your recent
+    updates to your branch. In this case, GitHub will suggest opening a pull
+    request for you. If not, switch to the tab "Pull request" in the top menu
+    and use the button "New pull request".
+
+    To create a pull request, make sure to select the default branch you used
+    when creating your branch as ``base`` (eg. ``branch-2.4``). Select your
+    branch as ``compare``.
+
+5. Write your pull request
+    After creating your pull request, GitHub will compare your branch to the
+    base branch and highlight all your proposed changes.
+
+    First, enter a title for your pull request. This title should make clear
+    what your pull request does. For example: "Fix PNG export", "Add panel to
+    tests", or "Document SVG backend".
+
+    Next, enter a description. Include some background about what your pull
+    request does and why you decided to write things the way you did. Also, link
+    to the issue your pull request is based on. To do so, use a keyword such as
+    "fixes," followed by the number of the issue. For example "Fixes #11479".
+    See `Linking a pull request to an issue using a keyword`_ in the
+    `GitHub documentation`_ for more information. Your description should also
+    include information about :ref:`tests <devguide_writing_tests>` and
+    :ref:`documentation <devguide_documentation>`, if applicable.
+
+6. Add more commits to your pull request
+    Once you have created a pull request, keep adding commits. A member of the
+    `Bokeh core team`_ will review your pull request and help you along the way.
```

Comment:
This part is sometimes not necessary. Maybe 
> will review your pull request,  and may request changes or additions. If so, they will help you along the way with any questions you may have. You can make new changes in your existing local branch, and push them to Github. The PR will update automatically (there is no need to open a new PR)

Also I think that last bit is important, some folks definitely try to close a PR just to add new commits.